compiler/rustc_codegen_llvm/src/context.rs

```diff
@@ -887,6 +888,9 @@ impl<'ll> CodegenCx<'ll, '_> {
             ifn!("llvm.dbg.declare", fn(self.type_metadata(), self.type_metadata()) -> void);
             ifn!("llvm.dbg.value", fn(self.type_metadata(), t_i64, self.type_metadata()) -> void);
         }
+
+        ifn!("llvm.ptrmask", fn(voidp, t_isize) -> voidp);
```

Comment:
Can someone check that this is valid signature? [llvm docs](https://llvm.org/docs/LangRef.html#llvm-ptrmask-intrinsic) give the following signature:
```llvm
declare ptrty llvm.ptrmask(ptrty %ptr, intty %mask) readnone speculatable
```
And the rust intrinsic has the following signature:
```rust
fn ptr_mask<T>(ptr: *const T, mask: usize) -> *const T;
```

Are all of these compatible with each other?

Reply:
The LLVM docs confuse me, here.  The semantic section says it's possible for the bitwidth of the mask to be different from the pointer size of the target, but that sounds to me like it'd be an overloaded intrinsic, and thus named something like `llvm.ptrmask.i64` the same way `llvm.cttz.*` and friends work.  But the name isn't documented the way those ones are...

Reply:
Yes, the name would be `llvm.ptrmask.p0i8.i64` etc.